### ISSUE TITLE
How much of C++ does this parse?

### ISSUE THREAD

**Issue by skunkiferous** [baszalmstra/header-parser#2]
Hi. I've spent an hour or two looking for code that can parse C++ (11) and output something that is usable to generate bindings automatically. [CastXML](https://github.com/CastXML/CastXML#readme) seems quite good, but does not parse templates, which I would also need. There might be other limitations that I am not aware of yet.

OTOH, it seems your project is aimed at parsing your own code, so it's not clear how much of C++ it understands. And since most questions about this topic mention that C++ is incredibly hard to parse, I would tend to assume your project doesn't support the full C++ syntax, unless explicitly specified.

Also, does it only extract info on code that is annotated?

My concrete goal is to parse the UnrealEngine headers, which are annotated, but I'm not sure everything I would want to use is annotated, like private fields ... I need enough info to not only know all the methods of the classes, but also the memory layout of the objects, to make them accessible from a "non C++" language.

**Comment by baszalmstra** [comment]
The idea behind this project is that it can correctly parse everything that is annotated and skip the rest. By skipping the rest it doesnt need to understand all of the c++ syntax. 

If you need to be able to parse unannotated code this project will not be of much help. 

You might be better off looking at Clang which can output an AST but requires a lot more boilerplate.
